Sources/Swift/Helper/SentryDispatchQueueWrapper.swift

```diff
@@ -46,6 +46,12 @@
         internalWrapper.dispatch(after: interval, block: block)
     }
 
+    public func dispatch(after interval: TimeInterval, workItem: DispatchWorkItem) {
+        internalWrapper.dispatch(after: interval) {
+            workItem.perform()
+        }
```

Comment:
`m`: I think we must also add this method to the internal wrapper, so we actually call `queue.asyncAfter(deadline: interval, execute: workItem)`. Then, otherwise we create another work item and schedule this one to the dispatch queue. When we cancel the DispatchWorkItem, I'm pretty sure the DispatchQueue checks the isCancelled flag and doesn't even run the work item anymore. With this approach we will call perform on a cancelled DispatchWorkItem, after cancelling it.

Reply:
Calling perform on a cancelled work item should return immediatley accoring to documentation.
```Swift
https://developer.apple.com/documentation/dispatch/dispatchworkitem/cancel()
```

AFAIK we cant use `DispatchWorkItem` in the ObjC implementation, so we can't move the call there, but we can access the queue in the first wrapper.

```Swift
public func dispatch(after interval: TimeInterval, workItem: DispatchWorkItem) {
    internalWrapper.queue.asyncAfter(deadline: .now() + interval, execute: workItem)
}
```

Reply:
Decent workaround thanks. Maybe add a comment why you have to access the internall queue.